exception/Stringify.js

```diff
@@ -5,44 +5,54 @@
  * Copyright © 2021 Extremely Heavy Industries Inc.
  */
 import {stripTags, trimToDepth} from '@xh/hoist/utils/js';
+import {omitBy, isNil} from 'lodash';
 
 /**
- * Stringify an error object (typically an exception) safely for submission to server.
- * This method will avoid circular references and will trim the depth of the stack.
+ * Serialize an error object safely for submission to server, or user display.
+ * This method will avoid circular references and will trim the depth of the object.
  *
- * @param {Object} errorObject - the error to serialize
+ * @param {Error} error
  * @return string
  */
-export function stringifyErrorSafely(errorObject) {
+export function stringifyErrorSafely(error) {
     try {
-        // Clone + protect against circularity, monstrosity with a general depth trim.
-        const err = trimToDepth(errorObject, 5);
 
-        // Delete noisy grails exception
-        const {serverDetails} = err;
+        // 1) Create basic structure.
+        // Raw Error does not have 'own' properties, so be explicit about core name/message/stack
+        // Order here intentional for serialization
+        let ret = {
+            name: error.name,
+            message: error.message
+        };
+        Object.assign(ret, error);
```

Comment:
To ensure I understand this - we might get a "Raw Error" with no "own" properties, so we explicitly extract name/message/stack - and in that case, does `Object.assign()` does nothing?

But we might also get an Error that's not "raw", in which case it would have additional properties installed - primary use case being exceptions created by our `Exception.js` methods.  In that case, any additional custom properties will get copied over with the assign.

Give a shout if that's *wrong* - but otherwise looks good, recording here only to confirm and have a note on it.

Reply:
great -- you got it